src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,112 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private int threshold = Math.round(DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    private int newCapacity = 0;
+    private Node<K, V>[] table;
+    private int size;
+
+    public MyHashMap() {
+        this.table = new Node[DEFAULT_INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        int index = 0;
+        if (key != null) {
+            index = Math.abs(key.hashCode()) % table.length;
+        }
+        while (true) {
+            if (table[index] == null) {
+                table[index] = new Node<>(objectHash(key), key, value, null);
+                size++;
+                if (size >= threshold) {
+                    resize();
+                }
+                return;
+            } else if (areValuesEqual(table[index].key, key)) {
+                table[index].value = value;
+                return;
+            } else {
+                index = (index + 1) % table.length;
+            }
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        int index = 0;
+        if (key != null) {
+            index = Math.abs(key.hashCode()) % table.length;
+        }
+        while (true) {
+            if (table[index] == null) {
+                return null;
+            } else if (areValuesEqual(table[index].key, key)) {
+                return table[index].getValue();
+            } else {
+                index = (index + 1) % table.length;
+            }
+        }
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private boolean areValuesEqual(K key, K current) {
+        return (current != null && current.equals(key))
+                || (current == null && key == null);
+    }
```

Comment:
name implies that we are comparing values but in fact these are keys, overall we can remove that method and use `Objects.equals()` instead since there is no restriction for it anymore

Reply:
Not resolved